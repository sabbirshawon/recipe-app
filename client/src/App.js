import React, { useState, useEffect, lazy, Suspense } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect,
} from 'react-router-dom';
import AuthContext from './contexts/auth-context';
import RecipesPage from './pages/Recipes';
import RecipeDetails from './pages/RecipeDetails';
import Login from './pages/Login';
import SignUp from './pages/Signup';
import AddRecipePage from './pages/AddRecipe';
import UserPage from './pages/UserPage';
import './App.scss';
import ForgetPassword from './pages/ForgetPassword';
import ResetPassword from './pages/ResetPassword';
import NoMatch from './pages/NoMatch';
import EditRecipe from './pages/EditRecipe';

const Navbar = lazy(() => import('./components/navigations/Navbar'));

const NavbarFallback = () => {
  return <div className="main-navigation"></div>;
};

function App() {
  const [token, setToken] = useState('');
  const [userId, setUserId] = useState('');

  useEffect(() => {
    const userInfoToken = JSON.parse(
      localStorage.getItem('userInfo')
    );
    const userIdLocal = JSON.parse(localStorage.getItem('userId'));
    if (userInfoToken && userIdLocal) {
      setToken(userInfoToken);
      setUserId(userIdLocal);
    }
  }, []);

  const login = (token, userId) => {
    setToken(token);
    setUserId(userId);
  };

  const logout = () => {
    setToken(null);
    setUserId(null);
    localStorage.removeItem('userInfo');
  };

  return (
    <Router>
      <AuthContext.Provider
        value={{
          token,
          userId,
          login,
          logout,
        }}
      >
        <Suspense fallback={<NavbarFallback />}>
          <Navbar />
        </Suspense>
        <main className="main-content">
          <Switch>
            {token && <Redirect from="/login" to="/" />}
            {token && <Redirect from="/registration" to="/" />}
            {!token && <Route path="/login" component={Login} />}
            {!token && (
              <Route path="/registration" component={SignUp} />
            )}
            <Route path="/" exact component={RecipesPage} />
            <Route path="/add-recipe" component={AddRecipePage} />
            <Route
              path="/recipes/:recipeId"
              exact
              component={RecipeDetails}
            />
            <Route
              path="/recipes/edit/:recipeId"
              component={EditRecipe}
            />
            <Route path="/users/:username" component={UserPage} />
            <Route
              path="/forget-password"
              component={ForgetPassword}
            />
            <Route path="/reset/:token" component={ResetPassword} />
            <Route component={NoMatch} />
          </Switch>
        </main>
      </AuthContext.Provider>
    </Router>
  );
}

export default App;
