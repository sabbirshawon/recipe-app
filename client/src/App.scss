.main-content {
  margin: 4rem 2.5rem;
}

.error_msg_wrap {
	background: red;
	padding: 15px;
	h2 {
		color: #fff;
	}
}

.success_msg_wrap {
	background: green;
	padding: 15px;
	h2 {
		color: #fff;
	}
}