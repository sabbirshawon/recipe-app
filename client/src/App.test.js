import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import App from './App';

afterEach(cleanup);

// making a snapshot
it('renders', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
