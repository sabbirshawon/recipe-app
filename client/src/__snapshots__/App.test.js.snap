// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="main-navigation"
  />
  <main
    class="main-content"
  >
    <div
      class="recipes_wrapper"
    >
      <div
        class="recipe_add_action"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained tooltip addRecipeBtn MuiButton-containedSecondary"
          tabindex="0"
          title="Login to add you recipe"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Add your recipe
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="recipes"
        data-testid="recipes"
      >
        <div
          class="loading"
        >
          <div
            class="loader"
            style="width: 120px; height: 120px;"
          />
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
